File: project-euler/problem17.py
one1 = ["one","two","three","four","five","six","seven","eight","nine"];
ten1 = ["ten","eleven","twelve","thirteen", "fourteen", "fifteen" , "sixteen", "nineteen", "seventeen", "eighteen"];
ten2 = ["twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"];


def hundred(x): return one(x/100)+"hundred" + (("end" + ten(x%100)) if x%100 > 0 else "");
def ten(x):
	if (x == 0): return 0;
	elif (x < 10): return one(x);
	elif (x < 20): return ten1[x-10];
	else: return ten2[(x/10)-2] + one(x%10);

def one(x): 
	if (x == 0): return "";
	return one1[x-1];

def count(x):
	print 1;

if __name__ == "__main__":
	sol = 0;
	for i in range(1,1000):
		if (i < 100): x = (ten(i));
		else: x = (hundred(i));
		sol += len(x);
		#sol += count(i);
	
	print sol + len("onethousand");

File: project-euler/problem17.cpp
#include <cstdio>
#include <cctype>
#include <cmath>
#include <cstdlib>

#include <string>
#include <iostream>
#include <vector>
#include <map>
#include <set>
#include <algorithm>

#define pb push_back
#define fi(it, v) for (typeof((v).begin()) it = (v).begin(); it != (v).end(); ++it)
#define fit(v) fi(it, v)
#define f(i,n) for(int i = 0; i < (n); ++i)

#define fs first
#define sc second
#define mp make_pair

#define all(v) (v).begin(), (v).end()
#define ll long long int
#define vi vector<int>
#define vii vector<vi >
using namespace std;

"one","two","three","four","five","six","seven","eight","nine"



int one(int x) {
 
}
int ten(int x){
    
        switch(x){
        case 1: return 3; //ten  
        case 2: return 3; //eleven
        case 3: return 5; //tw
        case 4: return 4; //four
        case 5: return 3; //five
        case 6: return 3; //six
        case 7: return 5; //seven
        case 8: return 5; //eight
        case 9: return 4; //nine
    } 
}

int hundred(int x){
    
    return one(x) + 7;   
}

int main(){
    
}
